includes/gateway/class-omise-payment-internetbanking.php

```diff
@@ -77,102 +77,47 @@ public function payment_fields() {
-				$charge = $this->sale( array(
-					'amount'      => $this->format_amount_subunit( $order->get_total(), $order->get_order_currency() ),
-					'currency'    => $order->get_order_currency(),
-					'description' => apply_filters('omise_charge_params_description', 'WooCommerce Order id ' . $order_id, $order),
-					'source'      => array( 'type' => $_POST['omise-offsite'] ),
-					'return_uri'  => add_query_arg( 'order_id', $order_id, site_url() . "?wc-api=omise_internetbanking_callback" ),
-					'metadata'    => array_merge( $metadata, array(
-						/** override order_id as a reference for webhook handlers **/
-						/** backward compatible with WooCommerce v2.x series **/
-						'order_id' => version_compare( WC()->version, '3.0.0', '>=' ) ? $order->get_id() : $order->id
-					) )
-				) );
-
-				$order->add_order_note( sprintf( __( 'Omise: Charge (ID: %s) has been created', 'omise' ), $charge['id'] ) );
-
-				switch ( $charge['status'] ) {
-					case 'pending':
-						$this->attach_charge_id_to_order( $charge['id'] );
-
-						$order->add_order_note( sprintf( __( 'Omise: Redirecting buyer out to %s', 'omise' ), esc_url( $charge['authorize_uri'] ) ) );
-
-						/** backward compatible with WooCommerce v2.x series **/
-						if ( version_compare( WC()->version, '3.0.0', '>=' ) ) {
-							$order->set_transaction_id( $charge['id'] );
-							$order->save();
-						} else {
-							update_post_meta( $order->id, '_transaction_id', $charge['id'] );
-						}
-
-						return array (
-							'result'   => 'success',
-							'redirect' => $charge['authorize_uri'],
-						);
-						break;
-
-					case 'failed':
-						throw new Exception( $charge['failure_message'] . ' (code: ' . $charge['failure_code'] . ')' );
-						break;
-
-					default:
-						throw new Exception(
-							sprintf(
-								__( 'Please feel free to try submit your order again or contact our support team if you have any questions (Your temporary order id is \'%s\')', 'omise' ),
-								$order_id
-							)
-						);
-						break;
-				}
-			} catch ( Exception $e ) {
-				wc_add_notice(
-					sprintf(
-						wp_kses(
-							__( 'Seems we cannot process your payment properly:<br/>%s', 'omise' ),
-							array( 'br' => array() )
-						),
-						$e->getMessage()
-					),
-					'error'
-				);
+			if ( 'pending' == $charge['status'] ) {
+				$order->add_order_note( sprintf( __( 'Omise: Redirecting buyer out to %s', 'omise' ), esc_url( $charge['authorize_uri'] ) ) );
 
-				$order->add_order_note(
-					sprintf(
-						__( 'Omise: Payment failed, %s', 'omise' ),
-						$e->getMessage()
-					)
+				return array (
+					'result'   => 'success',
+					'redirect' => $charge['authorize_uri'],
 				);
-
-				return;
 			}
+
+			return $this->payment_failed(
+				sprintf(
+					__( 'Please feel free to try submit your order again or contact our support team if you have any questions (Your temporary order id is \'%s\')', 'omise' ),
```

Comment:
Please feel free to try submitting your order again, or contact our support team if you have any questions (Your temporary order id is \'%s\')